watchtower-plugin/src/dbm.rs

```diff
@@ -216,28 +216,29 @@ impl DBM {
         &self,
         tower_id: TowerId,
         user_id: UserId,
+        subscription_start: Option<u32>,
+        subscription_expiry: Option<u32>,
     ) -> Option<RegistrationReceipt> {
-        let mut stmt = self
-            .connection
-            .prepare(
-                "SELECT available_slots, subscription_start, subscription_expiry, signature
-                    FROM registration_receipts 
-                    WHERE tower_id = ?1 AND subscription_expiry = (SELECT MAX(subscription_expiry) 
-                        FROM registration_receipts 
-                        WHERE tower_id = ?1)",
-            )
-            .unwrap();
+        let mut query = "SELECT available_slots, subscription_start, subscription_expiry, signature FROM registration_receipts WHERE tower_id = ?1 AND (subscription_start >=?2 OR ?2 is NULL) AND (subscription_expiry <=?3 OR ?3 is NULL)".to_string();
```

Comment:
This feels pretty hacky, especially when there is no boundary, well be passing `(AND NULL is NULL) AND (NULL is NULL)` which will be valid, but feels pretty unnecessary.

It'll be best to find a way to define the query and params list depending on whether `subscription_start` and `subscription_expiry` were passed